src/SourceBuild/content/Directory.Build.props

```diff
@@ -50,11 +50,12 @@
     <BuildRid Condition="'$(BuildRid)' == ''">$([System.Runtime.InteropServices.RuntimeInformation]::RuntimeIdentifier)</BuildRid>
 
     <HostRid Condition="'$(HostRid)' == ''">$(BuildRid)</HostRid>
-    <TargetRid Condition="'$(TargetRid)' == ''">$(BuildRid)</TargetRid>
+    <TargetRid Condition="'$(TargetRid)' == ''">$(BuildRid.Substring(0, $(BuildRid.LastIndexOf('-'))))-$(TargetArchitecture)</TargetRid>
```

Comment:
Rather than substrings, we don't we just set the TargetRid based on the TargetOS and TargetArch?

Reply:
This relates to what the user needs to specify to the build command.

I've been going for specifying the arch:
```
./build.sh --use-mono-runtime --source-build -- /p:TargetArchitecture=ppc64le
```
And the OS gets determined automatically, e.g. `linux` is used for a portable build on Linux.

Reply:
I'm not sure the `TargetOS` works in the non-portable build case (which also builds):
```
./build.sh --use-mono-runtime --source-build -- /p:TargetArchitecture=ppc64le /p:PortableBuild=false
```
-> this now builds `ubuntu.18.04-ppc64le` when cross-building from `ubuntu.18.04-x64`.

Reply:
Agreed that the user should specify less. But above we already determine the TargetOS (as `linux`). Why derive it again from the BuildRid? Is the BuildRid in this case non-portable?

Reply:
The `TargetRid` that is assigned here is only used in the non-portable case.

Reply:
As I suspected, `TargetOS` can't be used to determine non-portable rids. See these comments: https://github.com/dotnet/source-build/issues/4143#issuecomment-1948852515 and https://github.com/dotnet/source-build/issues/4143#issuecomment-1951950083.